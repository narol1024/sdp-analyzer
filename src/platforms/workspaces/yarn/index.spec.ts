/**
 * This is a sample test suite.
 * Replace this with your implementation.
 */
import { readWorkspacesDependencies, analyze } from './index';

describe('Workspaces Tests', () => {
  it('should be that the package.json cannot be found', () => {
    expect(() => {
      readWorkspacesDependencies('./error-path');
    }).toThrow(new Error(`Cannot find package.json on ./error-path.`));
  });
  it('should be that the package.json can be found', () => {
    expect(readWorkspacesDependencies('./src/platforms/workspaces/yarn/fixture')).toMatchSnapshot();
  });
  it('can analyze for local packages', async () => {
    await expect(analyze('./src/platforms/workspaces/yarn/fixture')).resolves.toMatchSnapshot();
  });
});
