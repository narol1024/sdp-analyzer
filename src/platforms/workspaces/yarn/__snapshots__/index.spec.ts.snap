// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workspaces Tests can be analyzed for the local packages 1`] = `
Array [
  Object {
    "fanIn": 1,
    "fanOut": 0,
    "instability": 0,
    "label": "Stable",
    "name": "a",
  },
  Object {
    "fanIn": 1,
    "fanOut": 1,
    "instability": 0.5,
    "label": "Normal",
    "name": "b",
  },
  Object {
    "fanIn": 1,
    "fanOut": 2,
    "instability": 0.6666666666666666,
    "label": "Flexible",
    "name": "c",
  },
  Object {
    "fanIn": 1,
    "fanOut": 3,
    "instability": 0.75,
    "label": "Flexible",
    "name": "d",
  },
  Object {
    "fanIn": 1,
    "fanOut": 4,
    "instability": 0.8,
    "label": "Instable",
    "name": "e",
  },
]
`;

exports[`Workspaces Tests should read a valid package 1`] = `
Array [
  Object {
    "dependencies": Object {},
    "name": "a",
  },
  Object {
    "dependencies": Object {
      "a": "^1.0.0",
    },
    "name": "b",
  },
  Object {
    "dependencies": Object {
      "a": "^1.0.0",
      "b": "^1.0.0",
    },
    "name": "c",
  },
  Object {
    "dependencies": Object {
      "a": "^1.0.0",
      "b": "^1.0.0",
      "c": "^1.0.0",
    },
    "name": "d",
  },
  Object {
    "dependencies": Object {
      "a": "^1.0.0",
      "b": "^1.0.0",
      "c": "^1.0.0",
      "e": "^1.0.0",
    },
    "name": "e",
  },
]
`;
